### PR TITLE
UI tests for reset password

### PR DESCRIPTION
**Pull request by paurakhsharma**
## Description
UI test for testing reset password function

## Related Issue
#31116 

## Motivation and Context
first step for acceptance tests for https://github.com/owncloud/core/pull/31092#pullrequestreview-111927009

## How Has This Been Tested?
run tests locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: tests/acceptance/features/bootstrap/WebUILoginContext.php [hunk 1]
```diff
@@ -207,7 +208,31 @@ public function itShouldBePossibleToLogin($shouldOrNot, $username, $password) {
...
+	 * @When the user requests the password reset link using the webUI
+	 * 
+	 * @return void
+	 */
+	public function theUserRequestsThePasswordResetLinkUsingTheWebui() {
+		$this->loginPage->requestPasswordReset($this->getSession());
+	}
+
+	/**
+	 * @Then the a message with this text should be displayed on the webUI:
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/WebUILoginContext.php]: s/Then the a message/Then a message

**Review by phil-davis — COMMENTED** [review]
Great stuff - just 1 little comment.
The next challenge is how to extract the link from the email text and then browse to it and actually enter a new password.

**Comment by paurakhsharma** [comment]
@phil-davis I have added some more tests, can you review it.

**Review by phil-davis — COMMENTED** [review]
This looks really good. I just made some small comments about names of methods...

File: tests/acceptance/features/webUILogin/resetPassword.feature [hunk 2]
```diff
@@ -0,0 +1,37 @@
+@webUI @insulated @disablePreviews
+
+Feature: send password reset email
+As a user
+I want to reset my password
+So that I can login my account with new password
```
[the remark this answers is not in the document]
Comment (paurakhsharma) [reply tests/acceptance/features/webUILogin/resetPassword.feature]: Thanks for the review and suggestions.
I will fix it right away.

**Review by phil-davis — APPROVED** [review]

**Comment by phil-davis** [comment]
@paurakhsharma please make a backport PR to ``stable10``

**Comment by paurakhsharma** [comment]
backport in #31192

**Pull request merged**